File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
       unordered_map<char,int> mp_s; 
       unordered_map<char,int> mp_ts; 
        if(s.size()!=t.size())
           return false;
        for(int i =0;i<s.size();i++){
         mp_s[s[i]]++;
         mp_ts[t[i]]++;
        }   
        for( char c : s){
            if(mp_s[c]!=mp_ts[c])
                 return false;
        }
        return true ;
        
    }
};

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
       Map<Character,Integer> mp_s = new HashMap<> (); 
       char[] ss=s.toCharArray();
       
        if(s.length()!=t.length())
           return false;
        for( char c :ss){
            mp_s.put(c,mp_s.getOrDefault(c,0)+1);
        }
        for(int i =0;i<s.length();i++){
         char c = t.charAt(i);
         if(!mp_s.containsKey(c) ||mp_s.get(c)==0){
            return false;
         }
         mp_s.put(c,mp_s.get(c)-1);
        }   
        
        return true ;
    }
}